integration/query_logical_test.go

```diff
@@ -0,0 +1,298 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+)
+
+func TestQueryLogicalAnd(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setupWithOpts(t, &setupOpts{providers: []shareddata.Provider{shareddata.Scalars}})
+
+	for name, tc := range map[string]struct {
+		filter      bson.D
+		expectedIDs []any
+		err         mongo.CommandError
+	}{
+		"And": {
+			filter: bson.D{{
+				"$and",
```

Comment:
`bson.D{{"$and", something}}` is shared across test cases, so we can not repeat it: https://github.com/FerretDB/FerretDB/blob/6e8a7773a9867f76e982ed9229594d6c899e8c46/integration/query_comparision_test.go#L308

Reply:
> `bson.D{{"$and", something}}` is shared across test cases, so we can not repeat it:
> 
> https://github.com/FerretDB/FerretDB/blob/6e8a7773a9867f76e982ed9229594d6c899e8c46/integration/query_comparision_test.go#L308

@AlekSi  sorry what did you mean?

Reply:
Fixed